provisioning/device/samples/readme.md

```diff
@@ -0,0 +1,101 @@
+# Provisioning Device Client Sample - Microsoft Azure IoT SDK for .NET
+
+## Overview
+
+_This documentation is preliminary and subject to change._
+
+This folder contains samples demonstrating the steps required to dynamically associate devices with IoT Hubs using the Microsoft Azure IoT Device Provisioning Service.
+
+To ensure that only authorized devices can be provisioned, two device attestation mechanisms are supported by the service: one based on X509 certificates and another based on Trusted Platform Module (TPM) devices. In both cases, public/private key authentication will be performed during provisioning.
+
+_Preview only:_ The SDK currently supports two communication protocols: HTTP and AMQP over TLS.
+
+## Device provisioning in a nutshell
+
+Provisioning is achieved by using a single call to the `ProvisioningDeviceClient.RegisterAsync()` API specifying the IDScope (unique for each Provisioning Service deployment), a `SecurityClient` and a `ProvisioningTransportHandler`:
+
+```C#
+    ProvisioningDeviceClient provClient = ProvisioningDeviceClient.Create(s_idScope, security, transport);
+    DeviceRegistrationResult result = await provClient.RegisterAsync();
+    if (result.Status != ProvisioningRegistrationStatusType.Assigned) 
+    {
+    Console.WriteLine($"ProvisioningClient AssignedHub: {result.AssignedHub}; DeviceID: {result.DeviceId}");
+    }
+```
+
+### Provisioning devices using X509 certificate-based attestation
+
+Devices must have access to a single certificate with a private key. The private key can be hidden from the application using PCKS11 Hardware Security Modules.
```

Comment:
Might want to verify the PKCS11 support with Simon.

Reply:
/cc @simonporter 

Windows handles PKCS11 the same way as a normal certificate except that the private key never leaves the device.

Reply:
Nit: this line has the K and the C backwards.